### ISSUE TITLE
GPD Win 3 - GamepadUI screen rotated

### ISSUE THREAD

**Issue by thezeagar** [ChimeraOS/chimeraos#534]
I wanted to load Chimera OS on my GPD Win 3 1165G7 model, so see about giving it a more straight forward console experience. I'd seen videos of it running on the Win 3, but so far I've had little luck.

After finally getting it to install (had to use 2022.06.02 for the system to boot after install), the new SteamOS GamePad UI is rotated 90 degrees counter-clockwise. I know this is a somewhat normal issue when installing different OS's on the Win 3 as it's a portrait screen; but the installer, terminal, and desktop modes all default to the correct orientation, it's just the new GamePad UI.

I found the device-quirks file on the system, referenced [here](https://github.com/ChimeraOS/gamescope-session/blob/main/usr/share/gamescope-session/device-quirks), and it looks like my device's product name, G1618-03 is listed, and it's suppose to detect that and rotate the screen? This does not seem to be activating.

Is there anything I need to or can do to get GamePad UI into the correct orientation?

Let me know if I can provide any additional information to help.

Thank you

**Comment by pastaq** [comment]
We're actively working on this device. It turns out that the DSI connector that it uses doesn't have hardware support for screen rotation, so we need to patch gamescope to use a software rotation solution. This also affects the OXP Gundam. We have a subthread on the discord in the help channel where we are actively posting the most recent updates/troubleshooting tips of you'd like to stay the most up to date on it.

**Comment by ruineka** [comment]
I'm working on this almost every day trying to learn new things, or find out more information to help get these "unique" devices supported. I made some recent discoveries as well as got more information from an Intel employee regarding hardware rotation which answered one question, but has created many more. I opened a ticket at the gamescope repo so those who run into this issue can share their experiences to hopefully bring notice to this.

TL;DR
Intel is known to not be great at rotating the display 90 or 270 degrees, but can rotate 0 and 180 just fine.

Follow the report here:
https://github.com/Plagman/gamescope/issues/819

**Comment by ruineka** [comment]
Oh, and I don't have confirmation on this, but I think the devices that ship with proper MiPi DSI to eDP adapters work around the hardware rotation limitation on intel devices. Our devices here use the MiPi DSI connector directly and it's exposed as `DSI-1` to gamescope. I think this is why other devices that show up as `eDP` and use the same 11th/12th gen Intel processors work.
